### ISSUE TITLE
Smart RTL with only few waypoints for a arduplane.

### ISSUE THREAD

**Issue by ninja-zx11** [ArduPilot/ardupilot#21903]
## Feature request

I would like to initiate a feature request of implementing just few points for smart RTL (like a copter) on arduplane. 

Lets say you are flying at around 300mts over the passage between the high mountains ( 2000m) and you accidently got behind the mountain. If using terrain data, plane would loose all its battery to achieve 2000+ meters. But if there is a smart RTL, it would quickly follow the path back.

I am aware of the memory limitation on the boards but if we can get only few points for smart RTL, it would be very useful.

May be only  few points would be useful  just to come back LOS.
**First few points for smart RTL and then switch to normal RTL like copter**.  In that way you can never loose your plane if accidently flying behind a high mountain.

**Platform**

[ * ] Plane

**Comment by timtuxworth** [comment]
I have personally had several crashes, and destroyed 2 planes, when RTL decided to fly through or outside a clearly marked geofence in order to fly home after a "failsafe". I have marked geofences around mountains, buildings, physical fences and trees. Fixed wing RTL will fly right into them. A smart RTL would be a good start to addressing this.

**Comment by ninja-zx11** [comment]
Exactly. I crashed my nicely built Mini Talon last month because of that:(

**Comment by cbf123** [comment]
I think this is going to be very similar to https://github.com/ArduPilot/ardupilot/issues/2380

I don't think memory is necessarily a limitation since you already need a decent chunk of memory to handle terrain data in the first place.

**Comment by ninja-zx11** [comment]
Coming here after a break from flying for few years and wondering if this idea or something similar has been implemented on Arduplane or not? Thanks.

**Comment by LupusTheCanine** [comment]
I don't think so, at least for fully automatic SmartRTL. IIRC the big issues with plane SmartRTL are initial turnaround (the only direction that can be reasonably assumed safe is up and most UAV planes lack performance for Immelman turn) and cornering as planes typically cut sharp corners much more than copters.

Return path start mission item was recently implemented to allow for smarter "backtracking" (technically it is skipping mission items to return path in a smarter way than do land start allows with fewer mission items).

I haven't tried to use bendy ruler on plane so I don't know if it works with fences.

PS I tried to workout an efficient way of preemptively avoiding fences on a plane and covering all fence cases is very hard and sometimes it may even force a plane into a more serious breach.
